controllers/front.go

```diff
@@ -0,0 +1,19 @@
+package controllers
+
+import (
+	"encoding/json"
+	"io"
+	"net/http"
+)
+
+func RegisterControllers() {
+	uc := newUserController()
+
+	http.Handle("/users", *uc)
+	http.Handle("/users/", *uc)
+}
+
+func encodeResponseAsJSON(data interface{}, w io.Writer) {
+	enc := json.NewEncoder(w)
+	enc.Encode(data)
```

Comment:
In this case, you could do `json.NewEncoder(w).Encode(data)`, however you should be checking for errors.

You can either:
```
if err := json.NewEncoder(w).Encode(data); err != nil {
    log.Fatal("fatal json error:", err)
}
```
which is usually not ideal as your program will die there, but may be appropriate in some cases.

The better option would be to change the return signature of encodeResponseAsJSON to return an error and then just do; `return json.NewEncoder(w).Encode(data)` and check the error when you call the function.